src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -173,6 +176,14 @@ export function CurrencySearch({
     filteredTokens.length === 0 || (debouncedQuery.length > 2 && !isAddressSearch) ? debouncedQuery : undefined
   )
 
+  // timeout token loader after 3 seconds
```

Comment:
Say why:
`// Timeout token loader after 3 seconds to avoid hanging in a loading state.`

Reply:
Done